pkg/apis/planetscale/v2/vitesscluster_methods.go

```diff
@@ -69,3 +69,13 @@ func (image *MysqldImage) Flavor() string {
 		return ""
 	}
 }
+
+func (externalOptions *ExternalVitessClusterUpdateStrategyOptions) Storage() bool {
```

Comment:
The options struct isn't all about resource changes, so we should add some context:

```suggestion
func (externalOptions *ExternalVitessClusterUpdateStrategyOptions) AllowStorageChanges() bool {
```

Actually maybe we should keep this general so we don't end up duplicating it with slightly different search terms:

```suggestion
func (externalOptions *ExternalVitessClusterUpdateStrategyOptions) ResourceChangesAllowed(resource corev1.ResourceName) bool {
```

Reply:
I had tons of issues with naming in these API changes so thanks for offering suggestions; I was very stuck. Also, changing this method to take in any resource is solid, done.